pkg/vcs/linux.go

```diff
@@ -211,32 +218,63 @@ func (ctx *linux) addMaintainers(com *Commit) {
 	if len(com.CC) > 2 {
 		return
 	}
-	list := ctx.getMaintainers(com.Hash, false)
-	if len(list) < 3 {
-		list = ctx.getMaintainers(com.Hash, true)
+	mtrs := ctx.getMaintainers(com.Hash, false)
+	if len(mtrs) < 3 {
+		mtrs = ctx.getMaintainers(com.Hash, true)
 	}
-	com.CC = email.MergeEmailLists(com.CC, list)
+	com.Recipients = mtrs
 }
 
-func (ctx *linux) getMaintainers(hash string, blame bool) []string {
+func (ctx *linux) getMaintainers(hash string, blame bool) []RecipientInfo {
 	// See #1441 re --git-min-percent.
 	args := "git show " + hash + " | " +
 		filepath.FromSlash("scripts/get_maintainer.pl") +
-		" --no-n --no-rolestats --git-min-percent=20"
+		" --git-min-percent=20"
 	if blame {
 		args += " --git-blame"
 	}
 	output, err := osutil.RunCmd(time.Minute, ctx.git.dir, "bash", "-c", args)
 	if err != nil {
 		return nil
 	}
-	var list []string
-	for _, line := range strings.Split(string(output), "\n") {
-		addr, err := mail.ParseAddress(line)
+	lines := strings.Split(string(output), "\n")
+	return ParseMaintainers(lines)
+}
+
+func ParseMaintainers(lines []string) []RecipientInfo {
+	if len(lines) < 1 {
+		return []RecipientInfo{}
+	}
+	reType := regexp.MustCompile(`\(.*?\)`)
+	var mtrs []RecipientInfo
+	listFlag := false // check if there are lists besides linux-kernel
+	for _, line := range lines {
+		mtr := RecipientInfo{}
+		reType := reType.FindString(line)
+		addr, err := mail.ParseAddress(strings.Replace(line, reType, "", 1))
 		if err != nil {
 			continue
 		}
-		list = append(list, strings.ToLower(addr.Address))
+		mtr.RecipientType = reType
+		mtr.RealName = addr.Name
+		mtr.Email = addr.Address
+		if strings.Contains(mtr.RecipientType, "reviewer") {
+			mtr.ToCc = false
+		} else if strings.Contains(mtr.RecipientType, "(open list)") {
+			continue
```

Comment:
Why do we ignore all (open list) entries? Won't we ignore some useful lists?

Reply:
I understood that LKML is the only one that comes in the format "[name] <[email]> (open list)" and the other ones come in the format "[name] <[email]> (open list: [description])". So we'd ignore LMKL because we want to add it after checking if there are any other lists.

Reply:
Since linux-kernel is handled specially anyway, perhaps it would make more sense to check for `linux-kernel@vger.kernel.org` instead of `(open list)`?

Reply:
Yes, that makes sense, I'll change it